packages/orchestration/src/exos/chain-hub.js

```diff
@@ -591,19 +591,23 @@ export const makeChainHub = (zone, agoricNames, vowTools) => {
       return undefined;
     },
     /**
-     * @param {string} partialId CAIP-10 account ID or a Cosmos bech32 address
+     * @param {AccountIdArg | string} account CAIP-10 account ID or a
+     *   CosmosChainAddress or bech32
      * @returns {AccountId}
      * @throws {Error} if chain info not found for bech32Prefix
      */
-    resolveAccountId(partialId) {
-      const parsed = parseAccountId(partialId);
-      if ('namespace' in parsed) {
-        // It is already fully qualified
-        return /** @type {AccountId} */ (partialId);
+    resolveAccountId(account) {
+      if (typeof account === 'string') {
```

Comment:
Please nest less. If you just check for `'value' in` first and handle that case with a return, then after the value will be narrowed to string and you don't have to even touch the other lines.

Reply:
We can't check for `'value' in account` until we establish that it's not a string.

Reply:
Oh, but if I make the conditional be `typeof account !== 'string'`, I can simplify as you suggested.  one sec....

Reply:
> LGTM but please squash that into the earlier commit. It's a fixup to something you're proposing to add to master. That it changed in the PR is not relevant to the history of master and adds noise to the blame history.

I think I can't do this (this time) without a forced push.  Given your "verbal" LGTM, I'm going to presume that's acceptable for this PR.

> Similarly the "chore: narrow type" should be squashed into the commit that had the wider guard.

ditto

> Check out https://github.com/tummychow/git-absorb to make it easy to automatically commit fixups

That'll be useful the next time.